lib/grp.gi

```diff
@@ -5070,11 +5073,24 @@ end);
 #M  HasSolvableFactorGroup(<G>,<N>)   test whether G/N is solvable
 ##
 InstallGlobalFunction(HasSolvableFactorGroup,function(G,N)
-local s,l;
+local gen, D;
+
+  if HasIsSolvableGroup(G) and IsSolvableGroup(G) then
+    return true;
+  fi;
   Assert(2,IsNormal(G,N) and IsSubgroup(G,N));
-  s := DerivedSeriesOfGroup(G);
-  l := Length(s);
-  return IsSubgroup(N,s[l]);
+  D := G;
```

Comment:
It does make sense in that case to check whether the last term of the derived series of G is in N, even if it might be slower in rare situations. Anyway, added to the commit, force pushed.